.github/workflows/plugin.yaml

```diff
@@ -0,0 +1,42 @@
+name: kubectl plugin
+
+on:
+  push:
+    branches:
+      - "main"
+    tags:
+      - "controller-*"
+
+jobs:
+  release-plugin:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      
+      - name: Convert to semver tag
+        run: |
+          echo "SEMVER_TAG<<EOF" >> $GITHUB_ENV
+          echo ${{github.ref_name}} | sed 's/controller-//' >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          version: latest
+          args: release --rm-dist
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GORELEASER_CURRENT_TAG: ${{ env.SEMVER_TAG }}  
+      
+      - name: Update new version in krew-index
+        uses: rajatjindal/krew-release-bot@v0.0.43
```

Comment:
Please use SHA for the versions and not sem ver.

Reply:
I could be reading this wrong but [I think goreleaser enforces semver?](https://goreleaser.com/limitations/semver/#semantic-versioning)

Reply:
so for krew-release-bot it would 92da038bbf995803124a8e50ebd438b2f37bbbb0 for v0.0.43

Checkout 2541b1294d2704b0964813337f33b291d3f8596b

setup-go 84cbf8094393cdc5fe1fe1671ff2647332956b1a

Reply:
Think this thread is out dated since the `GORELEASER_CURRENT_TAG` variable is no longer used in the most recent version of this PR.